### PR TITLE
feat: isolate network addresses between testnet 

### PR DESCRIPTION
**Pull request by greatertomi**
# Checklist

- [x] JIRA - LW-6813
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

Previously, the address book content is shared between preview and preprod. This PR isolates the address book content based on the network. It uses `Cardano.NetworkMagics` instead of `Cardano.NetworkId`

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ✅ [test report](https://lace-qa:8PP5wBaDV6UoXj@d1terqjryk7mu9.cloudfront.net/linux/chrome/437/5288469953/index.html) for [188b35e8](https://github.com/input-output-hk/lace/pull/126/commits/188b35e8f69b1558917c147d6020d6e28176233d)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 19     | 0      | 0       | 0     | 19    | ✅     |

### CODE REVIEW

File: apps/browser-extension-wallet/src/utils/validators/address-book.ts [hunk 1]
```diff
@@ -66,3 +72,29 @@ export const isValidAddressPerNetwork = ({
   address: string;
   network: Wallet.Cardano.NetworkId;
 }): boolean => !address || validateAddrPerNetwork[network](address);
+
+export const checkForDuplicateAddress = (
```
Comment (rhyslbw) [inline apps/browser-extension-wallet/src/utils/validators/address-book.ts:76]:
```suggestion
export const hasAddressBookItem = (
```

File: apps/browser-extension-wallet/src/utils/validators/__tests__/address-book.test.ts [hunk 2]
```diff
@@ -199,4 +202,29 @@ describe('Testing address book validator', () => {
       expect(addressBook.validateTestnetAddress('DdzFF')).toEqual(false);
     });
   });
+
+  describe('checkAddressForDuplicate', () => {
```
Comment (rhyslbw) [inline apps/browser-extension-wallet/src/utils/validators/__tests__/address-book.test.ts:206]:
```suggestion
  describe('hasAddressBookItem', () => {
```

File: apps/browser-extension-wallet/src/utils/validators/__tests__/address-book.test.ts [hunk 3]
```diff
@@ -199,4 +202,29 @@ describe('Testing address book validator', () => {
...
+    const mockAddressList: AddressBookSchema[] = Array.from({ length: 4 }, (_v, i) => ({
+      id: i + 1,
+      address: `addr_test${i + 1}`,
+      name: `test wallet ${i + 1}`,
+      network: Cardano.NetworkMagics.Preprod
+    }));
+
+    test('has an already existing name', () => {
```
Comment (rhyslbw) [inline apps/browser-extension-wallet/src/utils/validators/__tests__/address-book.test.ts:214]:
```suggestion
    test('has item with same name', () => {
```

File: apps/browser-extension-wallet/src/utils/validators/__tests__/address-book.test.ts [hunk 4]
```diff
@@ -199,4 +202,29 @@ describe('Testing address book validator', () => {
...
+      expect(
+        addressBook.checkForDuplicateAddress(mockAddressList, { name: 'test wallet 1', address: 'addr_test14' })[0]
+      ).toBe(true);
+    });
+    test('has an already existing address', () => {
```
Comment (rhyslbw) [inline apps/browser-extension-wallet/src/utils/validators/__tests__/address-book.test.ts:219]:
```suggestion
    test('has item with same address', () => {
```

File: apps/browser-extension-wallet/src/utils/validators/__tests__/address-book.test.ts [hunk 5]
```diff
@@ -199,4 +202,29 @@ describe('Testing address book validator', () => {
...
+      expect(
+        addressBook.checkForDuplicateAddress(mockAddressList, { name: 'test wallet 15', address: 'addr_test2' })[0]
+      ).toBe(true);
+    });
+    test('has a unique name and address', () => {
```
Comment (rhyslbw) [inline apps/browser-extension-wallet/src/utils/validators/__tests__/address-book.test.ts:224]:
```suggestion
    test('doesn't have an item with same name and address', () => {
```

File: packages/e2e-tests/src/features/SettingsPagePopup.feature [hunk 6]
```diff
@@ -205,12 +205,12 @@ Feature: General Settings - Popup View
     Then I see information about empty address book
 
   @LW-5472 @Testnet
-  Scenario: Popup View - Settings - Addresses are shared and saved when switching Test Preprod to Test Preview
+  Scenario: Popup View - Settings - Addresses are not shared when switching Test Preprod to Test Preview
```
Comment (rhyslbw) [inline packages/e2e-tests/src/features/SettingsPagePopup.feature:208]:
```suggestion
  Scenario: Popup View - Settings - Addresses entered in Preprod context are not available when switching to Preview
```

**Review by rhyslbw — CHANGES_REQUESTED** [review]

File: apps/browser-extension-wallet/src/utils/validators/address-book.ts (hunk 1 above)
Comment (greatertomi) [reply apps/browser-extension-wallet/src/utils/validators/address-book.ts]: fixed on 0da2e3c

File: apps/browser-extension-wallet/src/utils/validators/__tests__/address-book.test.ts (hunk 2 above)
Comment (greatertomi) [reply apps/browser-extension-wallet/src/utils/validators/__tests__/address-book.test.ts]: fixed on 0da2e3c

File: apps/browser-extension-wallet/src/utils/validators/__tests__/address-book.test.ts (hunk 3 above)
Comment (greatertomi) [reply apps/browser-extension-wallet/src/utils/validators/__tests__/address-book.test.ts]: fixed on 0da2e3c

File: apps/browser-extension-wallet/src/utils/validators/__tests__/address-book.test.ts (hunk 5 above)
Comment (greatertomi) [reply apps/browser-extension-wallet/src/utils/validators/__tests__/address-book.test.ts]: fixed on 0da2e3c

File: packages/e2e-tests/src/features/SettingsPagePopup.feature (hunk 6 above)
Comment (greatertomi) [reply packages/e2e-tests/src/features/SettingsPagePopup.feature]: fixed on 0da2e3c

**Comment by gabriela-ponce** [comment]
@greatertomi Here are a few issues/inconsistencies on this PR that I was not able to reproduce in the main branch

1 - Remove an address from the list, then complete the fields to add a new one and press "Enter".
**Result:** The "Address Removed" notification is displayed again at the bottom.
Check [this](https://drive.google.com/file/d/19KGAATgrItnljnPRCb-mtNXjoYNF7x-F/view?usp=drive_link) video

2 -  Remove an address from the list. Click on another address and press "Enter".
**Result:** The "Address Removed" notification is displayed again at the bottom. Changes are not applied.
Check [this](https://drive.google.com/file/d/1l-R_LQLzdKjy8Qyp-fJyi4kwXWhxnlb2/view?usp=drive_link) video

3 - Remove an address from the list. Click on another address and press "Edit". On the edition screen, change the name and press "Enter".
**Result:** The "Address Removed" notification is displayed again at the bottom. Changes are not applied.
Check [this](https://drive.google.com/file/d/1qONNmEcnjpG1JTdUg4-nmXikDNf24FtH/view?usp=drive_link) video

**Review by rhyslbw — APPROVED** [review]
Great work @greatertomi :rocket:

**Review by dmitrii-gaico — APPROVED** [review]

**Review by gabriela-ponce — APPROVED** [review]

**Pull request merged**